### ISSUE TITLE
Identify how to deal with filtering known clientside exceptions not solvable from triggering an exception email

### ISSUE THREAD

**Issue by evilaliv3** [globaleaks/GlobaLeaks#1382]
Given the new exception handler implemented on the GLClient it would be important to implement a small feature to keep track clientdie of the filtering of some exceptions.

Basically there are known exceptions like https://github.com/globaleaks/GlobaLeaks/issues/1381 that would need to be filtered until we would proparly understand how to fix them, or for life if we are bugs that are triggered by interrupting the app that we would be able to recognixe and fix.

basically this would involve simply to understand which kind of hashing apply to the stacktrace in order to filter ow know exceptions that we want to filter.

things will be complicated given the fact that lines of the exceptions change during time so that we will be probably in possibility to filter exceptions simply by family (their title-description string)

**Comment by fpietrosanti** [comment]
I think that's a gmail filter configuration to be done if it's annoying, without additional overhead, assuming that any unhandled exception is a buggy condition at least to be catched/handled at application level.
That's because we'd likely be jnterested in perceiving/knowing how many exception trigger, from where and when?

Sent from mobile

> On 30/ago/2015, at

**Comment by evilaliv3** [comment]
yep but also in the backend we have some irrilevant exceptions that we are not able to handle and it's normal to have in case of interruptions and on such exceptions the good action is simply to filter them

**Comment by evilaliv3** [comment]
not filtering such actions woulkd expose the whistlebloweer to possible privacy leak (as any exception handled this way export at least the timing of where and when the whistleblower was on the app) and so when it's not useful for us and we can minimize we would have for sure to do it.

**Comment by fpietrosanti** [comment]
Which kind of backend unhandled exception need to be filtered rather than handled?

Sent from mobile

**Comment by evilaliv3** [comment]
i dont know why are you digging into backend execption now given the fact that this ticket is specifically on the client, but i hope in the end you will end in using instrument properly and not faking of being doing that :)

by the way: in the end i hope all exceptions will b handled (sometimes handling means simply ignoring them as there is no action to solve a brutal close of the browser while twisted is waiting something from the user) but if you remember we had various finish() something exceptions that were reaching our exception handler and given that twisted has tons of conditions where this happen instead of fixing them all with (with a simple "if exception of this kind ignore"  among all the code) in the meantime i've simple written a filter one time in the main unhandled exception handler.

the exceptions handled this way are only three (https://github.com/globaleaks/GlobaLeaks/blob/master/backend/globaleaks/utils/mailutils.py#L261) for the backend and the new one for the client and the clientside one of ticket 1381 is going to require such the same threatment i think (but i've still to evaluate it; but in my first analysis i was not able to solve it by not making it happen and there is no action to solve it at all except eventually skip/filter it)

**Comment by evilaliv3** [comment]
Closing as discarded since client exception notification was deprecated for many reasons and especially privacy concerns.
